### PR TITLE
[15.0] [MIG] stock_picking_invoice_link

### PR DESCRIPTION
**Pull request by antonyht27**
Standard migration to version 15.0.

### CODE REVIEW

**Review by florenciafrigieri2 — APPROVED** [review]
Functional review ok, working as expected.

File: stock_picking_invoice_link/README.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fstock--logistics--workflow-lightgray.png?logo=github
+    :target: https://github.com/OCA/stock-logistics-workflow/tree/14.0/stock_picking_invoice_link
+    :alt: OCA/stock-logistics-workflow
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/stock-logistics-workflow-14-0/stock-logistics-workflow-14-0-stock_picking_invoice_link
```
Comment (celm1990) [inline stock_picking_invoice_link/README.rst]: Please replace `14.0` to `15.0` on all reference

**Review by celm1990 — CHANGES_REQUESTED** [review]

**Comment by celm1990** [comment]
Is possible add fields for record count??(on picking and invoice)

image


image

File: stock_picking_invoice_link/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    picking_ids = fields.Many2many(
+        comodel_name="stock.picking",
+        string="Related Pickings",
+        store=True,
+        compute="_compute_picking_ids",
+        help="Related pickings "
+        "(only when the invoice has been generated from a sale order).",
+    )
+    
+    delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_count_ids')
```
Comment (celm1990) [inline stock_picking_invoice_link/models/account_move.py]: please compute fields into same function `_compute_picking_ids`

**Review by celm1990 — CHANGES_REQUESTED** [review]

**Comment by grindtildeath** [comment]
@antonyht27 Can you please cherry-pick the changes from https://github.com/OCA/stock-logistics-workflow/pull/816 ?

**Review by nicomacr — APPROVED** [review]
LGTM +1

File: stock_picking_invoice_link/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            )
+        )
+
+    def _prepare_invoice_line(self, **optional_values):
+        vals = super()._prepare_invoice_line(**optional_values)
+        stock_moves = self.get_stock_moves_link_invoice()
+        # Invoice returned moves marked as to_refund
+        if float_compare(self.qty_to_invoice, 0.0, precision_rounding=self.currency_id.rounding) < 0:
+            stock_moves = stock_moves.filtered(lambda m: m.to_refund and not m.invoice_line_ids)
+        vals["move_line_ids"] = [(4, m.id) for m in stock_moves]
```
Comment (celm1990) [inline stock_picking_invoice_link/models/sale_order.py:29]: This not working
I create a Sale Order by 10U
delivery 10 U and create invoice, picking and invoice are linked

Now Create a return and try create new invoice(is create as refund) but picking is not asociate with refund

image

No invoice related
image

**Review by celm1990 — CHANGES_REQUESTED** [review]

**Comment by florenciafrigieri2** [comment]
Hi @antonyht27 ! Are you going to continue with this PR? 
The suggestion about the returned picking-invoice is working correctly in V14.

**Comment by patel1008b** [comment]
Hello @antonyht27 thanks for the migration, Is this ready for merge and production use ??

**Comment by rousseldenis** [comment]
@antonyht27 Do you mind if we switch to https://github.com/OCA/stock-logistics-workflow/pull/1006 ?

**Comment by celm1990** [comment]
> @antonyht27 Do you mind if we switch to #1006 ?

Please close this PR, and continue working on other PR, Antony not working with us any more

**Pull request closed without merging**